packages/client/hmi-client/src/workflow/ops/model-comparison/tera-model-comparison.vue

```diff
@@ -222,9 +226,29 @@ function formatField(field: string) {
 	return result.charAt(0).toUpperCase() + result.slice(1);
 }
 
-// function submitGollmQuestion() {
-// 	console.log(gollmQuestion.value);
-// }
+function updateImagesState(newImageId?: string, clearImages = false) {
+	const state = cloneDeep(props.node.state);
+	if (newImageId) state.comparisonImageIds.push(newImageId);
+	if (clearImages) state.comparisonImageIds = [];
+	emit('update-state', state);
+}
```

Comment:
Avoid using disjunctions and optionals in function signatures, it leads to confusion and bug, and is generally a bad design.

People should be able to read the function signature and reason with what it is doing, you cannot do that if the parameters are self-contradictory.

If you want to squeeze multiple features into a single function, the pattern I would recommend is to pass in an operator type and a data structure. So in your case it would be:

```
function updateImageState(operationType: string, imageIds: string[] | null = null) {
   ...
}
```